src/runtime_src/core/common/api/aie/xrt_graph.cpp

```diff
@@ -321,6 +367,39 @@ read_port(const std::string& port_name, void* value, size_t bytes)
 
 } // namespace xrt
 
+////////////////////////////////////////////////////////////////
+// xrt_aie_event C++ API implmentations (xrt_aie.h)
+////////////////////////////////////////////////////////////////
+namespace xrt { namespace aie {
+
+event::
+event(const xrt::device& device)
+  : impl(std::move(create_event(device)))
+{}
+
+int 
+event::
+start_profiling(int option, const std::string& port1_name, const std::string& port2_name, uint32_t value) const
+{
+  return xdp::native::profiling_wrapper("xrt::aie::event::read_profiling", [=]{return (impl->start_profiling(option, port1_name, port2_name, value));});
```

Comment:
I think you want to capture the strings by reference.  And,
This would be impossible to debug with a break point, better format like this:
```suggestion
  return xdp::native::profiling_wrapper("xrt::aie::event::read_profiling", [this, &port1, &port2, value] {
    return impl->start_profiling(option, port1_name, port2_name, value);
  });
```